geode-wan/src/distributedTest/java/org/apache/geode/internal/cache/wan/parallel/ParallelWANPersistenceEnabledGatewaySenderDUnitTest.java

```diff
@@ -2231,6 +2240,102 @@ public void testpersistentWanGateway_restartSenderWithCleanQueuesDelayed_expectN
     vm3.invoke(() -> WANTestBase.validateRegionSize(getTestMethodName(), 0));
   }
 
+
+  /**
+   * Enable persistence for region as well as GatewaySender and see if remote site receives all the
+   * events.
+   */
+  @Test
+  public void testPersistentPartitionedRegionWithGatewaySenderStartStopEventsDispatchedNoChangesInQueue() {
+    Integer lnPort = vm0.invoke(() -> WANTestBase.createFirstLocatorWithDSId(1));
+    Integer nyPort = vm1.invoke(() -> WANTestBase.createFirstRemoteLocator(2, lnPort));
+
+    createCacheInVMs(nyPort, vm2, vm3);
+
+    createCacheInVMs(lnPort, vm4, vm5);
+    vm4.invoke(() -> setNumDispatcherThreadsForTheRun(5));
+    vm5.invoke(() -> setNumDispatcherThreadsForTheRun(5));
+
+    vm4.invoke(() -> WANTestBase.createSender("ln", 2, true, 100, 10, false, true, null, false));
+    vm5.invoke(() -> WANTestBase.createSender("ln", 2, true, 100, 10, false, true, null, false));
+
+    vm4.invoke(() -> WANTestBase.createPersistentPartitionedRegion(getTestMethodName(), "ln", 1,
+        100, isOffHeap()));
+    vm5.invoke(() -> WANTestBase.createPersistentPartitionedRegion(getTestMethodName(), "ln", 1,
+        100, isOffHeap()));
+
+
+    vm2.invoke(() -> WANTestBase.createPersistentPartitionedRegion(getTestMethodName(), null, 1,
+        100, isOffHeap()));
+    vm3.invoke(() -> WANTestBase.createPersistentPartitionedRegion(getTestMethodName(), null, 1,
+        100, isOffHeap()));
+
+    vm4.invoke(() -> WANTestBase.doPuts(getTestMethodName(), 1000));
+
+
+    vm4.invoke(() -> WANTestBase.stopSender("ln"));
+    vm5.invoke(() -> WANTestBase.stopSender("ln"));
+
+    logger.info("Stopped all the senders.");
+
+    // wait for senders to stop
+    vm4.invoke(waitForSenderNonRunnable());
+    vm5.invoke(waitForSenderNonRunnable());
+
+    createReceiverInVMs(vm2, vm3);
+
+    AsyncInvocation inv1 = vm4.invokeAsync(startSenderRunnable());
+    AsyncInvocation inv2 = vm5.invokeAsync(startSenderRunnable());
+
+    try {
+      inv1.join();
+      inv2.join();
+    } catch (InterruptedException e) {
+      e.printStackTrace();
+      fail();
+    }
+
+    logger.info("Waiting for senders running.");
+    // wait for senders running
+    vm4.invoke(waitForSenderRunnable());
+    vm5.invoke(waitForSenderRunnable());
+
+    logger.info("All the senders are now running...");
+
+    AsyncInvocation inv3 = vm4.invokeAsync(stopSenderRunnable());
+    AsyncInvocation inv4 = vm5.invokeAsync(stopSenderRunnable());
+
+    try {
+      inv3.join();
+      inv4.join();
+    } catch (InterruptedException e) {
+      e.printStackTrace();
+      fail();
+    }
```

Comment:
This would be better as:
```
    AsyncInvocation<Void> inv3 = vm4.invokeAsync(stopSenderRunnable());
    AsyncInvocation<Void> inv4 = vm5.invokeAsync(stopSenderRunnable());

    inv3.get();
    inv4.get();
```
to avoid using the deprecated method `AsyncInvocation.join()` or using `fail()` which does not provide any details about why the test failed in the error message.

I'm also a little concerned that the test may become flaky if for some reason there is a delay between the two async invocations completing, due to GC or something, which would then make the assertion on line 2322 fail.